File: java/Algorithms-daily/src/com/sm/daily/ContaminationSpread.java
package com.sm.daily;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

/*
 * Given n number of persons in a park. One of them is having a virus. But we don't know whom. 
 * Also, the position of all persons is given. A contaminated person can spread it up to d distance.
 *  When the best case (Spread is minimum) and the worst case(Spread is maximum) would occur?e.g.
   N=5
   Position=[1, 3, 5, 9, 14]
   d=5
   Remark: There is vagueness in question, position does not mean distance.
   If we consider position as distance, what is the reference point for this distance.
   So lets assume people are in line in park and each position meant the unit of measure from the origin. 
 */
public class ContaminationSpread {

	public static void main(String[] args) {
		List<Integer> positions = List.of(1, 3, 5, 9, 14);
		findMinMaxSpread(positions, 5);
	}
	
	public static void findMinMaxSpread(List<Integer> pos, int distance) {
		List<Integer> spread = new ArrayList<>();
		for(int i = 0; i < pos.size() - 1; ++i) {
			int spreadCount = 0;
			for(int j = i + 1; j < pos.size(); ++j) {
				if(Math.abs(pos.get(i) - pos.get(j)) <= distance) {
					++spreadCount;
				}
			}
			spread.add(spreadCount);
		}
		System.out.println("max possible spread:" + Collections.max(spread));
		System.out.println("min possible spread: " + Collections.min(spread));
	}
}

File: rust/contamination_spread.rs


fn main(){
     let mut pos = vec![1i32,1, 3, 5, 9, 14 ];
     contamination_spread(pos.as_mut_slice(), 5);
}

fn contamination_spread(v : &mut [i32], dist: i32){

   let mut cache = Vec::new();
   for i in 0..v.len() - 1{
       let mut spread = 0;
       for j in i+1..v.len(){
           if (v[i] - v[j]).abs() <= dist{
               spread += 1; 
           } 
      }
      cache.push(spread);
   }  
   let min = cache.iter().min();
   match min{
       Some(min) => println!( "Min value: {}", min ),
       None      => println!( "Vector is empty" )
   }
   let max = cache.iter().max();
   match max{
       Some(max) => println!( "Max value: {}", max ),
       None      => println!( "Vector is empty" )
   }
}
